information/specification.md

```diff
@@ -0,0 +1,826 @@
+
+Over time, selecting entries randomly in this way will tend to _limit_ the
+amount of dust that accumulates in the UTxO set.
+
+#### Principle 2: Change Management
+
+As mentioned in the [Goals](#goals) section, it is desirable that coin
+selection algorithms, over time, are able to create UTxO sets that have
+_useful_ outputs: outputs that will allow us to process future payments with a
+_reasonably small_ number of inputs.
+
+If for each payment request of value **v** we create a change output of
+_roughly_ the same value **v**, then we will end up with a distribution of
+change values that matches the typical value distribution of payment
+requests.
+
+#### Principle 3: Performance Management
+
+Searching the UTxO set for additional entries to _improve_ our change outputs
+is _only_ useful if the UTxO set contains entries that are sufficiently
+small enough. But it is precisely when the UTxO set contains many small
+entries that it is less likely for a randomly-chosen UTxO entry to push the
+total above the upper bound.
+
+### Cardinality
+
+The Random-Improve algorithm imposes the following cardinality restriction:
+
+  * Each entry from the [initial UTxO set](#initial-utxo-set) is used to pay
+    for _at most one_ output from the [requested output
+    list](#requested-output-list).
+
+As a result of this restriction, the algorithm will fail with a [UTxO Not
+Fragmented Enough](#utxo-not-fragmented-enough) error if the number of entries
+in the [initial UTxO set](#initial-utxo-set) is _smaller than_ the number of
+entries in the [requested output list](#requested-output-list).
+
+### State
+
+At all stages of processing, the algorithm maintains the following pieces of
+state:
+
+ 1. #### Available UTxO Set
+
+    This is initially equal to the [initial UTxO set](#initial-utxo-set).
+
+ 2. #### Accumulated Coin Selection
+
+    The accumulated coin selection is a [coin selection](#coin-selection) where
+    all fields are initially equal to the _empty set_.
+
+### Computation
+
+The algorithm proceeds in two phases.
+
+#### Phase 1: Random Selection
+
+In this phase, the algorithm iterates through each of the [requested
+outputs](#requested-output-list) in descending order of coin value, from
+largest to smallest.
+
+For each output of value **v**, the algorithm repeatedly selects entries at
+**random** from the [available UTxO set](#available-utxo-set), until the _total
+value_ of selected entries is greater than or equal to **v**. The selected
+entries are then _associated with_ that output, and _removed_ from the
+[available UTxO set](#available-utxo-set).
+
+This phase ends when _every_ output has been associated with a selection of
+UTxO entries.
+
+#### Phase 2: Improvement
+
+In this phase, the algorithm attempts to improve upon each of the UTxO
+selections made in the previous phase, by conservatively expanding the
+selection made for each output in order to generate improved change
+values.
```

Comment:
It could be nice to point out that this phase can not make the algorithm fails and is totally optional. As per the 3rd principle, failing to improve _probably_ means that there's no need to improve in the first place which is part of the beauty of this whole algorithm.